client/src/components/TransactionItemEditor/TransactionItemEditor.tsx

```diff
@@ -63,8 +72,19 @@ const TransactionItemEditor: React.FC<TargetItem> = ({
   const onSubmit: SubmitHandler<FormInputs> = (
     data: TransactionTransferData
   ) => {
-    // code to run on submit
-    editItem(id, data, pageType);
+    editItem(budgetId, categoryId, id, {
+      title: data.title,
+      amount: pageType === "expense" ? data.amount * -1 : data.amount,
+    });
+    editBudget(budgetData[0]._id, {
+      title: budgetData[0].title,
+      total: budgetData[0].total,
+      // NEED TO FIX THIS
```

Comment:
?

Reply:
Please ignore for now! This comment was meant for the part that is suppose to do the math for the budget's currentAmount whenever a user EDITS a transaction. The budgetData[0].currentAmount is problematic if the user decides to edit the transaction right after posting before a refetch is done to update the data. I'll have to see how I can solve this so that it's not an issue

Reply:
In fact, I may be able to fix this now with the new BE method you wrote for getting the particular budgets info. I'll get to it tonight :D